akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala

```diff
@@ -275,9 +333,21 @@ final class BackoffSupervisor(
   def onTerminated: Receive = {
     case Terminated(ref) if child.contains(ref) ⇒
       child = None
-      val restartDelay = calculateDelay(restartCount, minBackoff, maxBackoff, randomFactor)
-      context.system.scheduler.scheduleOnce(restartDelay, self, StartChild)
-      restartCount += 1
+      val maxNrOfRetries = strategy match {
+        case oneForOne: OneForOneStrategy ⇒ oneForOne.maxNrOfRetries
+        case _                            ⇒ -1
```

Comment:
I wonder if we should create a constant instead of using the magic `-1` number - might be OK like this though.

Reply:
I thought of that too. but since -1 was used in other places and it's meaning was mentioned in the comments, I didn't use a constant. So, should I change it?